src/core/server/legacy/legacy_service.ts

```diff
@@ -64,8 +68,8 @@ export class LegacyService implements CoreService {
   constructor(private readonly coreContext: CoreContext) {
     this.log = coreContext.logger.get('legacy-service');
   }
-
-  public async setup(deps: SetupDeps) {
+  public async setup() {}
+  public async start(setupDeps: SetupDeps, startDeps: StartDeps) {
```

Comment:
This is kinda weird, but I don't think this is avoidable unless KbnServer had a separate construction and starting phases. How much work would that be and do you think it's worth it?

Reply:
actually we can have a canonical signature here, I will correct.